dig.go

```diff
@@ -692,6 +692,36 @@ func isFieldOptional(f reflect.StructField) (bool, error) {
 	return optional, err
 }
 
+type group struct {
+	name    string
+	flatten bool
+}
+
+func parseGroup(f reflect.StructField) (group, error) {
+	tag := f.Tag.Get(_groupTag)
+	if tag == "" {
+		return group{}, nil
+	}
+
+	components := strings.Split(tag, ",")
+	var g group
+	for i, c := range components {
+		if i == 0 {
+			g.name = c
+			continue
+		}
```

Comment:
string.Split will always return a non-empty list so we can simplify this a bit:

```suggestion
	g := group{name: components[0]}
	for _, c := range components[1:] {
```